llvm/lib/Support/SHA1.cpp

```diff
@@ -307,3 +307,46 @@ std::array<uint8_t, 20> SHA1::hash(ArrayRef<uint8_t> Data) {
   memcpy(Arr.data(), S.data(), S.size());
   return Arr;
 }
+
+std::string SHA1::hashToString(ArrayRef<uint8_t> Data, StringRef Mapping) {
+  constexpr unsigned BitsPerChar = 6;
+  constexpr unsigned CombinationPerChar = 1 << BitsPerChar;
+  constexpr unsigned LoopIterations = 27;
```

Comment:
Why? Put an expression.
I have the feeling that just using `auto` everywhere is enough instead of `unsigned` all over the place.

Reply:
i am not a fan of using auto in this context.